scripts/pre_commit_linter.py

```diff
@@ -116,22 +117,67 @@ def _get_changed_filenames():
     return unstaged_files + staged_files
 
 
-def _get_all_files_in_directory(dir_path):
+def _filename_match(filename, sources):
+    """Checks if filname matches with source
```

Comment:
spelling: filename

Having said that, I can't actually understand what this means from the function name and docstring. How about calling this: _does_glob_pattern_include_filename(glob_pattern, filename) and using a docstring like:

    Checks whether 'filename' is described by 'glob_pattern'.